### ISSUE TITLE
Sorting DataTable with Inputs

### ISSUE THREAD

**Issue by chilligerchief** [carbon-design-system/carbon-components-svelte#1061]
Hello. I have a small DataTable with Inputs in each cell. 
Somewhat like this:

```
<DataTable 
    headers={[{key:"name", value:"name"}, { key: 'positionx', value: 'X' }, { key: 'positiony', value: 'Y'}]}
        sortable
>
    <span slot="cell" let:row let:cell style="width:100%; display:inline-block; cursor:pointer" >	
	<input type="text" bind:value={allrects[row.id][cell.key]}/>
	<TextInput  bind:value={allrects[row.id][cell.key]}/>
     </span>
</DataTable>
```

This works fine, unless I sort one column. The normal HTML input works as expected. The TextInput component changes all values in all columns to `undefined`. When changing the value outside of the table the value updates correctly. I guess it is some issue with the binding of the TextInput. However, I cannot get behind what it is exactly. For now I just use normal inputs as a workaround...

EDIT: I read, that the Input should display empty text if the value is undefined. However, in this case it still displays `undefined` for some reason...

**Comment by metonym** [comment]
Interesting – could you provide a minimal repro? Happy to investigate

**Comment by chilligerchief** [comment]
Thanks for looking into it. Here I have a minimal example:
```
<script>
    import {
      DataTable,
      TextInput
    } from "carbon-components-svelte"; 
    let rows = [
        {id:0, x:123, y:345},
        {id:1, x:124, y:346},
        {id:2, x:125, y:347}
    ]
</script>
<DataTable 
    headers={[{ key: 'x', value: 'X' }, { key: 'y', value: 'Y'}]}
    rows={rows}
        sortable
>
    <span slot="cell" let:row let:cell style="width:100%; display:inline-block; cursor:pointer" >	
	<input type="text" bind:value={rows[row.id][cell.key]}/>
	<TextInput  bind:value={rows[row.id][cell.key]}/>
     </span>
</DataTable>
```

**Comment by metonym** [comment]
Fixed in [v0.58.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.58.0)
